### DISCUSSION TITLE
Index Progress notifications via j-hui/fidget.nvim

### DISCUSSION THREAD

**Discussion by sitedyno** [phpactor/phpactor#2322] [Q&A] [answered]
This was previously working for me I believe before this commit (or around this time):
https://github.com/phpactor/language-server/commit/413b6580bef77bbb46b824760105dd0aec7919d7

Is there a way to re-enable this? I've looked at the documentation and through the code some but can't find anything obvious (to me).

**Comment by dantleech** [comment]
hi, thanks for the sponsorship!

i am using the same plugin and can confirm that it works.

- do you get "standard" indexing notifications? (i.e. messages in the window)

if you *do* get standard indexing notifications then somehow your client (nvim) is reporting that it does not support them (maybe fidget isn't being setup, i.e. with `require"fidget".setup{}`):

https://github.com/phpactor/phpactor/blob/5761c14126f68d9ff623e89816044e61770a9846/vendor/phpactor/language-server/lib/WorkDoneProgress/ClientCapabilityDependentProgressNotifier.php#L61-L65

if you don't then there is likely another issue as it does one or the other

**Reply by sitedyno** [reply, marked as answer]
I finally figured this out. 

I was using `lspconfig.util.on_setup` to add a hook so I didn't need to duplicate `on_attach = on_attach, capabilities = capabilities` for every `lspconfig.[server_name].setup()` call.

A change in `cmp-nvim-lsp` completely overwrites `capabilities.textDocument` wich clobbers the `window.workDoneProgress = true` & other stuff. `cmp-nvim-lsp` previously selectively overwrote what was provided to it: https://github.com/hrsh7th/cmp-nvim-lsp/commit/389f06d3101fb412db64cb49ca4f22a67882e469

I guess when you do `lspconfig.[server_name].setup({capabiltities = capabilities})` it does a deep merge (works for 'nvim-cmp-lsp` instructions) where as in the hook I was completely overwriting it 🤦‍♂️
